master/buildbot/process/factory.py

```diff
@@ -92,19 +92,13 @@ def __init__(self, source, configure="./configure",
                  configureEnv=None,
                  configureFlags=None,
                  reconf=None,
-                 compile=None,
-                 test=None,
-                 distcheck=None):
+                 compile=["make", "all"],
```

Comment:
I agree that this is bad coding style to initialize method argument with mutable objects.
http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments

in your case, None has a meaning, so you cannot use it as a default Marker, so usually what you do is:

```
class defaultMarker(object):
    pass

[...]
def __init__(self, compile=defaultMarker):
    if compile is defaultMarker:
        compile = ['make', 'all']

```